### PR TITLE
Update tests to run only necessary source compilation

### PR DESCRIPTION
**Pull request by amygimma**
https://github.com/reconhub/reportfactory/issues/55

MAKES THINGS PASS!

I tried to update the examples in `factory_template_with_examples.zip` but for some reason it wasn't working. We'll update that as a team later?

### CODE REVIEW

**Comment by amygimma** [comment]
also makes testing faster bc we don't need to compile all reports for these tests :)

File: tests/testthat/test_new_factory.R [hunk 1]
```diff
@@ -110,7 +110,8 @@ test_that("working directory unchanged", {
   ## new_factory with move_in = FALSE should not alter the working directory
   expect_identical(odir, getwd())
 
-  update_reports(quiet = TRUE, factory = x)
+  test_report <- grep(list_reports(x), pattern = "contacts", value = TRUE)[1]
```
Comment (thibautjombart) [inline tests/testthat/test_new_factory.R]: For consistency, would `list_report(pattern = "contacts")` work here as well?

File: tests/testthat/test_validate_factory.R [hunk 2]
```diff
@@ -16,7 +16,9 @@ test_that("A new_factory is valid", {
     length(test$errors) == 0L && length(test$warnings) == 0L
   }
   expect_true(no_probs(validate_factory(x)))
-  update_reports(quiet = TRUE, factory = x)
+  test_report <- grep(list_reports(x), pattern = "contacts", value = TRUE)[1]
```
Comment (thibautjombart) [inline tests/testthat/test_validate_factory.R]: For consistency, would `list_report(pattern = "contacts")` work here as well?

Comment (amygimma) [reply tests/testthat/test_validate_factory.R]: ahh right, yes

File: tests/testthat/test_new_factory.R [hunk 3]
```diff
@@ -110,7 +110,7 @@ test_that("working directory unchanged", {
   ## new_factory with move_in = FALSE should not alter the working directory
   expect_identical(odir, getwd())
 
-  test_report <- grep(list_reports(x), pattern = "contacts", value = TRUE)[1]
+  test_report <- list_report(pattern = "contacts")
```
Comment (thibautjombart) [inline tests/testthat/test_new_factory.R]: I think it need to be plural - `list_report**s**`?

**Pull request merged**